tmt/steps/provision/mrack.py

```diff
@@ -946,25 +977,35 @@ def is_ready(self) -> bool:
     def _create(self, tmt_name: str) -> None:
         """ Create beaker job xml request and submit it to Beaker hub """
 
-        data: dict[str, Any] = {
-            'tmt_name': tmt_name,
-            'hardware': self.hardware,
-            'name': f'{self.image}-{self.arch}',
-            'os': self.image,
-            'group': 'linux',
-            }
-
-        if self.whiteboard is not None:
-            data["whiteboard"] = self.whiteboard
-
-        if self.arch is not None:
-            data["arch"] = self.arch
+        data = CreateJobParameters(
+            tmt_name=tmt_name,
+            hardware=self.hardware,
+            arch=self.arch,
+            os=self.image,
+            name=f'{self.image}-{self.arch}',
+            whiteboard=self.whiteboard or tmt_name,
+            beaker_job_owner=self.beaker_job_owner)
 
         try:
             response = self.api.create(data)
-        except ProvisioningError as mrack_provisioning_err:
-            raise ProvisionError(
-                f"Failed to create, response:\n{mrack_provisioning_err}")
+
+        except ProvisioningError as exc:
+            import xmlrpc.client
+
+            cause = exc.__cause__
+
+            if isinstance(cause, xmlrpc.client.Fault):
+                if 'is not a valid user name' in cause.faultString:
+                    raise ProvisionError(
+                        f"Failed to create Beaker job, job owner '{self.beaker_job_owner}' "
+                        "was refused as unknown.") from exc
+
+                if 'is not a valid submission delegate':
```

Comment:
Seems like the second if statement is missing `in cause.faultString`.
```suggestion
                if 'is not a valid submission delegate' in cause.faultString:
```

Reply:
Good catch, fixed in 399fd78c.